tests/ping.go

```diff
@@ -0,0 +1,50 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"fmt"
+
+	expect "github.com/google/goexpect"
+	"k8s.io/utils/net"
+
+	v1 "kubevirt.io/client-go/api/v1"
+)
+
+// PingFromVMConsole performs a ping through the provided VMI console.
+func PingFromVMConsole(vmi *v1.VirtualMachineInstance, ipAddr, prompt string) error {
+	pingString := "ping"
+	if net.IsIPv6String(ipAddr) {
+		pingString = "ping -6"
+	}
+	cmdCheck := fmt.Sprintf("%s %s -c 1 -w 5\n", pingString, ipAddr)
+	err := CheckForTextExpecter(vmi, []expect.Batcher{
+		&expect.BSnd{S: "\n"},
+		&expect.BExp{R: prompt},
+		&expect.BSnd{S: cmdCheck},
+		&expect.BExp{R: prompt},
+		&expect.BSnd{S: "echo $?\n"},
+		&expect.BExp{R: "0"},
+	}, 30)
+	if err != nil {
+		return fmt.Errorf("Failed to ping VMI %s, error: %v", vmi.Name, err)
```

Comment:
[also replied to the same issue raised here : https://github.com/kubevirt/kubevirt/pull/3479#discussion_r458291020]

I am not sure here, but don't you think the test logs will be more readable if  we only return the error here, and then format it in the `Expect`? or maybe add a more high-level message in the `Expect` formatting on top of this message?

If a test fails, the whole error object is printed including the message. I find it less convenient reading the error message from there, and always looking for the high level logs that are a part of the test logs. if I want to get into more details I will read the error message. 

as it is now, instead if a stand-by-it's-own high level message , and another more low-level message in the error object, you will have a long error message within the error object. 
another thing is that the high-level message won't show up in the tests summery, which I find also convenient to look for a quick debugging.(about the last one:  I am not sure the high level message will apear as well, but anyway it will be more readable, I think...)

I hope this is clear ...
What do you think?